docs/docs/create-source-plugin.md

```diff
@@ -100,3 +100,56 @@ package provides a set of helper functions to generate Node objects with the
 required fields. This includes automatically generating fields like node IDs
 and the `contentDigest` MD5 hash, keeping your code focused on data gathering,
 not boilerplate.
+
+## Advanced
+
+### Adding relationships between nodes
+
+Gatsby source plugins not only create nodes, they also create relationships between nodes that are exposed to GraphQL queries.
+
+There are two ways of adding node relationships in Gatsby: (1) hierarchical (parent-child) or (2) foreign-key based. 
+
+#### Hierarchical relationships
+
+An example of a hierarchical relationship would be a `transformer` plugin changing a file node's markdown string into a `childMarkdownRemark` node that is attached to that fileNode. The markdown node doesn't make sense outside of the file it belongs to. 
+
+When a parent node is deleted, Gatsby deletes all of the child nodes (and their child nodes, and so on). 
+
+*Creating the relationship*
+
+In order to create a parent/child relationship, when calling `createNode` for the child node the object that is passed in should have a `parent` key with the value set to the parent node's `id`. After this, call the `createParentChildLink` function exported inside `actions`.
+
+*Example*
+
+[Here's an example](https://github.com/gatsbyjs/gatsby/blob/1fb19f9ad16618acdac7eda33d295d8ceba7f393/packages/gatsby-transformer-sharp/src/on-node-create.js#L3-L25) from the `gatsby-transformer-sharp` source plugin.  
+
+#### Foreign-key relationships
+
+An example of a foreign-key relationship would be a Post that has an Author. 
+
+In this relationship, each object is a distinct entity that exists whether or not the other does, with independent schemas, and field(s) on each entity that reference the other entity -- in this case the Post would have an Author, and the Author might have Posts. The API of a service that allows complex object modelling, for example a CMS, will often allow users to relationships between entities and expose them through the API.
+
+When an object node is deleted, Gatsby _does not_ delete any referenced entities. 
+
+*Creating the relationship*
+
+Let's say you want to create a relationship between two nodes, which we'll call nodes A and B. 
+
+Before you pass Object A and Object B into `createNode` and create the nodes, you need to:
+
+1. create a field on Object A to hold the relationship to Object B. (You may want to name this field `relationships` for clarity). The value of this field should be an object.
```

Comment:
The pattern for having a object called "relationships" is pretty specific to Drupal since that's how Drupal handles things. It's pretty source dependent. Contentful always has named relationships. I don't think putting relationships in a sub-object should be the general pattern.

I'd continue first with your earlier Post/Author example and show how you'd create three nodes, 1 Author node and 2 Post nodes with the Post nodes having an `author` field pointing at the Author node's ID.